src/python_testing/TC_ACL_2_11.py

```diff
@@ -113,24 +123,47 @@ async def test_TC_ACL_2_11(self):
             care_struct = Clusters.AccessControl.Structs.AccessRestrictionEntryStruct(E1, C1, R1)
 
             cluster = ALL_CLUSTERS[C1]
-
             for restriction in R1:
                 restriction_type = restriction.type
                 ID1 = restriction.id
 
-                attribute = ALL_ATTRIBUTES[C1][ID1]
-                command = ALL_ACCEPTED_COMMANDS[C1][ID1]
-
                 if restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kAttributeAccessForbidden:
-                    await self.read_single_attribute_expect_error(cluster=cluster, attribute=attribute, error=Status.AccessRestricted, endpoint=E1)
+                    # if ID1 is null, it means it is a wildcard.  We need to read all attributes on the cluster
+                    if ID1 is NullValue:
```

Comment:
Followup: this might be clearer with a helper function that takes the cluster and attribute id and returns an iterator over the relevant attributes (one in the non-wildcard case), so you could do something like:
```
    for (attr_id, attribute) in getRelevantAttributes(C1, ID1):
        if attr_id in self.global_attributes:
            continue; # Or have getRelevantAttributes filter it out
        # Do the read_single_attribute_expect_error here
```

That way you only have the wildcard-expansion logic in one spot instead of copy/pasting it.